adds more tests for api.Symbols

xeno-by opened the pull request:
I overlooked some of the tests that should be added to the public API.
This commit fixes my mistake. Also see 027b00171c.

To reiterate the reasoning behind introduction of test methods into the API.
I'd argue that the API duplication w.r.t flags is trumped by unified interface
to tests and better encapsulation (especially since some of the flags have
different meaning depending on whether the flaggee is a term or a type).

xeno-by commented:
review by @adriaanm or @paulp

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/57/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/761/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/57/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/761/

jsuereth commented:
+1 to this from me.

xeno-by commented:
PLS REBUILD ALL

xeno-by commented:
the updated commit just adds the `isVarargs` method

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/64/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/768/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/64/

xeno-by commented:
also exposed `isSpecialized` as per Paul Butcher's request: http://groups.google.com/group/scala-internals/msg/ea2e4f6fd1afa1d9

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/770/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/770/

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/768/

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/69/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/69/

adriaanm commented on src/reflect/scala/reflect/api/Symbols.scala:
```diff
@@ -158,10 +158,22 @@ trait Symbols extends base.Symbols { self: Universe =>
+
+    /** Is this symbol a generated specialized member?
+     */
+    def isSpecialized: Boolean
```
confusing --> could also be interpreted as "annotated with @specialized"

xeno-by replied:
actually, both @specialized type parameters and generated specialized members get the SPECIALIZED flag. updated the pull request.

adriaanm commented on src/reflect/scala/reflect/api/Symbols.scala:
```diff
@@ -240,6 +252,25 @@ trait Symbols extends base.Symbols { self: Universe =>
+
+    /** Does this symbol represent a parameter that has a default value?
+     */
+    def isDefaultParam: Boolean
```
hasDefault? isParamWithDefault?

xeno-by replied:
I'd say that `isXXX` methods are preferable to `hasXXX`, because they are consistent with other `isXXX` methods.

xeno-by replied:
after discussing with Martin we decided to rename this to `isParamWithDefault`

adriaanm commented on src/reflect/scala/reflect/api/Symbols.scala:
```diff
@@ -286,6 +317,10 @@ trait Symbols extends base.Symbols { self: Universe =>
 
+    /** Does this method support variable length argument lists?
+     */
+    def isVarargs: Boolean
```
hasRepeatedParam?

xeno-by replied:
same argumentation as for `isDefaultParam` vs `hasDefault`

adriaanm commented:
I think we should discuss these additions during the next reflection meeting.

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/73/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/776/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/73/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/776/

odersky commented:
LGTM after rename

xeno-by committed:
adds more tests for api.Symbols

I overlooked some of the tests that should be added to the public API.
This commit fixes my mistake. Also see 027b00171c.

To reiterate the reasoning behind introduction of test methods into the API.
I'd argue that the API duplication w.r.t flags is trumped by unified interface
to tests and better encapsulation (especially since some of the flags have
different meaning depending on whether the flaggee is a term or a type).

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/81/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/784/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/81/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/784/
